upgrade-1.x-to-2.x.sh

```diff
@@ -0,0 +1,410 @@
+cp -a /etc/machine-id /mnt/state/
+
+# Copy hostname
+cp -a /etc/hostname /mnt/state/root-overlay/etc
+
+# Create some config dirs
+mkdir -p /mnt/state/root-overlay/etc/systemd/system/resin.target.wants
+mkdir -p /mnt/state/root-overlay/etc/resin-supervisor
+mkdir -p /mnt/state/root-overlay/etc/NetworkManager/system-connections
+
+# Create systemd files to start services
+wants_dir=/mnt/state/root-overlay/etc/systemd/system/resin.target.wants
+ln -s /lib/systemd/system/openvpn-resin.service $wants_dir
+ln -s /lib/systemd/system/prepare-openvpn.service $wants_dir
+ln -s /lib/systemd/system/resin-supervisor.service $wants_dir
+ln -s /lib/systemd/system/update-resin-supervisor.timer $wants_dir
+
+# Copy resin-supervisor config
+cp -a /etc/supervisor.conf /mnt/state/root-overlay/etc/resin-supervisor
+
+# resinOS 1.22 and 1.23 can have bad supervisor tags
+sed -i -e 's/@TARGET_TAG@/v2.8.3/' /mnt/state/root-overlay/etc/resin-supervisor/supervisor.conf
+
+# Copy docker config
+mkdir -p /mnt/state/root-overlay/etc/docker
+cp -a /etc/docker/* /mnt/state/root-overlay/docker
+
+# Copy dropbear config
+mkdir -p /mnt/state/root-overlay/etc/dropbear
+cp -a /etc/dropbear/* /mnt/state/root-overlay/dropbear
+
+# Create some /var dirs
+mkdir -p /mnt/state/root-overlay/var/lib/systemd
+mkdir -p /mnt/state/root-overlay/var/volatile/lib/systemd
+
+# Copy systemd var files
+cp -a /var/lib/systemd/* /mnt/state/root-overlay/var/lib/systemd
+
+# Ensure /mnt/boot is mounted
+mount ${root_dev}p1 /mnt/boot
+
+# Migrate wifi config to NetworkManager
+if grep service_home_wifi /mnt/boot/config.json >/dev/null; then
+    # Get wifi credentials
+    ssid=$(jq </mnt/boot/config.json '.["files"]."network/network.config"' | sed -e 's/.*Name = \([^\\"]*\).*/\1/')
+    psk=$(jq </mnt/boot/config.json '.["files"]."network/network.config"' | sed -e 's/.*Passphrase = \([^\\"]*\).*/\1/')
+
+    # Write NetworkManager setup
+    cat >/mnt/state/root-overlay/etc/NetworkManager/system-connections/resin-wifi <<EOF
+[connection]
+id=resin-wifi
+type=wifi
+
+[wifi]
+hidden=true
+mode=infrastructure
+ssid=$ssid
+
+[wifi-security]
+auth-alg=open
+key-mgmt=wpa-psk
+psk=$psk
+
+[ipv4]
+method=auto
+
+[ipv6]
+addr-gen-mode=stable-privacy
+method=auto
+EOF
+fi
+
+# Make /root/.docker
+mkdir -p /mnt/state/root-overlay/home/root/.docker
+
+# Touch openssl rnd file
+touch /mnt/state/root-overlay/home/root/.rnd
+
+# Restart docker
+systemctl start docker
+
+# Remount backup dir
+mount ${root_dev}p3 /tmp/backup
+
+IMAGE=resin/resinos:${TARGET_VERSION}-${DEVICE}
+CONTAINER=export-${TARGET_VERSION}
+BACKUPARCHIVE=/tmp/backup/newos.tar.gz
+FSARCHIVE=/mnt/data/newos.tar.gz
+
+# Create container for new version
+docker create --name ${CONTAINER} ${IMAGE} echo export
+
+# Export container
+docker export ${CONTAINER} | gzip > ${BACKUPARCHIVE}
+
+# Remove container
+docker rm ${CONTAINER}
+
+# Copy resin-data to backup partition before we wipe data
+cp /mnt/data/resin-data.tar.gz /tmp/backup/
+
```

Comment:
These steps above take quite a bit of time, and very little shown in the log. It would be good, if there was some logged info so one knows where things are. It feels like that those comments could actually be debug log items (similar to existing resinhup scripts), so they inform both when reading and when running the script. The current one sometimes hard to tell what line/step it actually is on.